tests/api/crud/test_alert_activations.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2024 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from datetime import datetime
+
+import pytest
+
+import server.api.crud.alert_activation
+
+
+@pytest.mark.parametrize(
+    "partition_interval, partition_datetime, expected_name, expected_partition_value, expected_expression",
+    [
+        (
+            "DAY",
+            datetime(2024, 10, 30),
+            "20241030",
+            "2024-10-31",
+            "DAY(activation_time)",
+        ),
+        (
+            "MONTH",
+            datetime(2024, 10, 30),
+            "202410",
+            "2024-11-01",
+            "MONTH(activation_time)",
+        ),
+        (
+            "YEARWEEK",
+            datetime(2024, 10, 30),
+            "202444",
+            "202445",
+            "YEARWEEK(activation_time, 1)",
```

Comment:
Given our discussion on ISO format and other formats, it may be useful to test an edge case where mode 1 of `YEARWEEK` is different than other modes (week 53 last year vs. week 1 this year).